PipelinesTasks/PRMetrics/src/task/src/git/gitInvoker.ts

```diff
@@ -81,20 +89,25 @@ export default class GitInvoker {
   private getTargetBranch (): string {
     this._logger.logDebug('* GitInvoker.getTargetBranch()')
 
-    const variable: string = Validator.validate(process.env.SYSTEM_PULLREQUEST_TARGETBRANCH, 'SYSTEM_PULLREQUEST_TARGETBRANCH', 'GitInvoker.getTargetBranch()')
+    const variable: string = Validator.validateVariable('SYSTEM_PULLREQUEST_TARGETBRANCH', 'GitInvoker.getTargetBranch()')
     const expectedStart: string = 'refs/heads/'
-    if (!variable.startsWith(expectedStart)) {
-      throw Error(`Environment variable SYSTEM_PULLREQUEST_TARGETBRANCH '${variable}' in unexpected format.`)
+    if (variable.startsWith(expectedStart)) {
+      const startIndex: number = expectedStart.length
+      return variable.substring(startIndex)
+    } else {
+      return variable
     }
-
-    const startIndex: number = expectedStart.length
-    return variable.substring(startIndex)
   }
 
   private getPullRequestId (): string {
     this._logger.logDebug('* GitInvoker.getPullRequestId()')
 
-    return Validator.validate(process.env.SYSTEM_PULLREQUEST_PULLREQUESTID, 'SYSTEM_PULLREQUEST_PULLREQUESTID', 'GitInvoker.getPullRequestId()')
+    const variable: string = Validator.validateVariable('BUILD_REPOSITORY_PROVIDER', 'GitInvoker.getPullRequestId()')
+    if (variable === 'GitHub') {
```

Comment:
`BUILD_REPOSITORY_PROVIDER` is equal to `GitHub` only for public GitHub instances. For GitHub Enterprise, it would be `GitHubEnterprise`

Reply:
Thanks, I've now updated this throughout